resources/2024-07-secretstore-feature-spec.md

```diff
@@ -0,0 +1,417 @@
+
+As a developer, I can reference an `Applications.Core/secretStores` in my `Applications.Datastores/*` or `Applications.Messaging/*` resource definition so that I no longer have to store secrets as plain text in the `properties.secrets` field of my resource for authentication, etc.
+
+### Detailed User Experience
+
+Step 1: Operator creates or references an existing `Applications.Core/secretStores` resource containing the secret data required for authenticating into resources that their developers may use.
+
+Creating a new `secretStore` resource: 
+
+```bicep
+resource authcreds 'Applications.Core/secretStores@2023-10-01-preview' = {
+  name: 'authcreds'
+  properties:{
+    application: application
+    // today the type is an enum, see https://github.com/radius-project/radius/pull/7816
+    type: 'generic'
+    data: {
+      'username': {
+        value: username
+      }
+      'password': {
+        value: password
+      }
+      'uri': {
+        value: uri
+      }
+      'connectionString': {
+        value: connectionString
+      }
+    }
+  }
+}
+```
+
+Referencing an existing `secretStore` resource:
+
+```bicep
+resource azurekeyvaultsecrets 'Applications.Core/secretStores@2023-10-01-preview' = {
+  name: 'existing-azurekeyvault-secrets'
+  properties:{
+    application: application
+    resource: 'secret-app-existing-secret'
+    // today the type is an enum, see https://github.com/radius-project/radius/pull/7816
+    type: 'generic'
+    data: {
+      'name': {
+        value: secret1
+      }
+      'version': {
+        value: 1
+      }
+      'encoding': {
+        value: 'base64'
+      }
+      'alias': {
+        value: secretalias
+      }
+    }
+  }
+}
+```
+
+Step 2: Operator references the `Applications.Core/secretStores` resource in their Recipe's `output` object to securely pass secrets as outputs to Radius that can be used in provisioning and deploying the resource using the Recipe.
+
+Bicep example:
+```diff
+param context object
+
+@description('The ID of the subnet where the Azure Cache for Redis will be deployed')
+param subnetID string = ''
+
+@description('The location of the Azure Cache for Redis')
+param location string = resourceGroup().location
+
+resource redis 'Microsoft.Cache/redis@2022-05-01' = {
+  name: 'redis-${uniqueString(context.resource.id)}'
+  location: location
+  properties: {
+    sku: {
+      capacity: 1
+      family: 'P'
+      name: 'Premium'
+    }
+    enableNonSslPort: true
+    minimumTlsVersion: '1.2'
+    subnetId: subnetID == '' ? null : subnetID
+  }
+}
+
+output result object = {
+  values: {
+    host: redis.properties.hostName
+    port: redis.properties.port
+    username: ''
+  }
+  secrets: {
++    password: {
++      valueFrom: {
++        secretRef: {
++          source: authcreds.id
```

Comment:
I missed this before. I don't believe bicep would allow you to reference a resource that is not defined within this file.